internal-services/catalog/file-updates-process-file-updates-task.yaml

```diff
@@ -0,0 +1,117 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: task-process-file-updates
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+      Update files in a Git repository
+  params:
+      - name: fileUpdates
+        type: string
+        description: fileUpdates json string
+  results:
+      - name: fileUpdatesInfo
+        description: fileUpdates detailed information
+      - name: fileUpdatesState
+        description: fileUpdates state
+  workspaces:
+      - name: temp
+        description: Temporary working directory
+  steps:
+      - name: perform-updates
+        image: quay.io/hacbs-release/release-utils@sha256:b850b1a4e93d16050eefdb967b7875c17d7af4e86ecb0b268674dc6c3e8d90b1
+        env:
+          - name: GITLAB_HOST
+            valueFrom:
+                secretKeyRef:
+                    name: file-updates-secret
+                    key: gitlab_host
+          - name: ACCESS_TOKEN
+            valueFrom:
+                secretKeyRef:
+                    name: file-updates-secret
+                    key: gitlab_access_token
+          - name: GIT_AUTHOR_NAME
+            valueFrom:
+                secretKeyRef:
+                    name: file-updates-secret
+                    key: git_author_name
+          - name: GIT_AUTHOR_EMAIL
+            valueFrom:
+                secretKeyRef:
+                    name: file-updates-secret
+                    key: git_author_email
+        script: |
+            #!/usr/bin/env sh
+            set -eo pipefail
+
+            # loading git and gitlab functions
+            . /home/utils/gitlab-functions
+            . /home/utils/git-functions
+
+            gitlab_init
+
+            # saves the fileUpdates raw json to a file
+            TEMP=$(mktemp).json
+            cat > $TEMP <<JSON
```

Comment:
This way bash does not have the change to remove or add quotes or escape chars where is should not. But I will give a try again.

Reply:
I don't think bash will add or remove anything when you run `echo "$(params.fileUpdates)"`. 

Or this is another option:

```sh
cat > $TEMP <<< "$(params.fileUpdates)"
```

I tried it with env vars and it worked fine even if there were double quotes in the strings (e.g. a json string). But here it might be different, because here the param substitution is done by tekton and you might end up with something like:
```sh
echo "{"a": "b"}"
```
which wouldn't work. So it needs testing.

Reply:
I tried  `echo "$(params.paths)" > $TEMP` and it broke exactly in the echo. I guess this is not a problem with the bash itself, but because this `$(params.xxx)` is a tekton "macro".

Reply:
` 68             cat > $TEMP <<< "$(params.paths)"` also broke the same way. It is now `params.paths`  because I am not using the full `fileUpdates` as param anymore.

The error FYI

```
+ TEMP=/tmp/tmp.m3yFeDDpF3.json
/tekton/scripts/script-0-d9cph: line 15: syntax error near unexpected token `('
```

Reply:
OK, thanks for testing it :)